easybuild/main.py

```diff
@@ -230,6 +230,11 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
     if 'CDPATH' in os.environ:
         del os.environ['CDPATH']
 
+    # When EB is run via `exec` the special bash variable $_ is not set
+    # So emulate this here to allow (module) scripts depending on that to work
+    if '_' not in os.environ:
+        os.environ['_'] = os.path.basename(sys.executable or 'python')
```

Comment:
Why the `basename`? It seems like the default value for `$_` is full path, so just use `sys.executable` (is there a scenario where that could be `None`?!)

```
$ python3 -c "import os; print(os.getenv('_'))"
/usr/bin/python3
$ python2 -c "import os; print(os.getenv('_'))"
/usr/bin/python2
```

Reply:
You are right. Changed that.